libraries/Microsoft.Bot.Builder.Dialogs.Rules/Input/NumberInput.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs.Rules.Input
+{
+    public class NumberInput<TNumber> : InputWrapper<NumberPrompt<TNumber>, TNumber> where TNumber : struct, IComparable<TNumber>
+    {
+        public TNumber MinValue { get; set; }
+        public TNumber MaxValue { get; set; }
+
+        public NumberInput()
+        {
+        }
+
+        protected override NumberPrompt<TNumber> CreatePrompt()
+        {
+            return new NumberPrompt<TNumber>(null, new PromptValidator<TNumber>(async (promptContext, cancel) =>
```

Comment:
@tomlm is this what you had in mind when we talked?

Reply:
yep

---
In reply to: [269795826](https://github.com/Microsoft/botbuilder-dotnet/pull/1569#discussion_r269795826) [](ancestors = 269795826)